tests/unit/extension/tools.js

```diff
@@ -0,0 +1,24 @@
+'use strict';// eslint-disable-line strict
+
+const assert = require('assert');
+
+const checkLimit = require('../../../lib/extension/tools').checkLimit;
+
+describe('checkLimit function', () => {
+    const tests = [
+        { input: [ 1000, 2500 ], output: 1000 },
+        { input: [ 5000, 2500 ], output: 2500 },
+        { input: [ 'lala', 2500 ], output: 2500 },
+        { input: [ 15000, 0 ], output: 15000 },
+        { input: [ 'lala', 0 ], output: 0 },
+        { input: [ 2000, undefined ], output: 2000 },
+        { input: [ 'lala', undefined ], output: undefined },
```

Comment:
what about cases such as 0,0 ?
I see that you test with an undefined or 0-defined limit, but as it is known as an invalid value for our implementation, shouldn't we ensure that this cant be done, or at least that the result yielded is valid for our system ?